### ISSUE TITLE
buildAuthenticatedFetch no longer supports custom fetch as a first argument

### ISSUE THREAD

**Issue by mrkvon** [inrupt/solid-client-authn-js#3810]
This is either bug or (regressed) feature request.

### Search terms you've used

`buildAuthenticatedFetch`

### Impacted package

Which packages do you think might be impacted by the bug ?

- [x] solid-client-authn-core

### Description

It looks like since v2, the library no longer supports passing custom fetch to `buildAuthenticatedFetch` (used to be first argument).  That can be considered a breaking change and feature regression. Please note that others depend directly on `buildAuthenticatedFetch`, and passing custom fetch can be very useful. In my case building authenticated request in Cypress.

### Expected result

It would be awesome if we can continue passing custom fetch to buildAuthenticatedFetch (e.g. as an option). I don't mind if API changes (e.g. passing fetch as optional property in second parameter).

**Comment by NSeydoux** [comment]
Thanks for reporting this, let me look into it.

**Comment by mrkvon** [comment]
FYI CSS guide [Automating authentication with Client Credentials](https://communitysolidserver.github.io/CommunitySolidServer/latest/usage/client-credentials/) depends on `buildAuthenticatedFetch`. It's the reason why I use it as well, both for sending authenticated requests in cypress and [css-authn](https://github.com/mrkvon/css-authn) library.

**Comment by michielbdejong** [comment]
> would it be possible to inverse the responsibilities and build the customized fetch from the authenticated one?

Yes, that would work for some use cases, but no, unfortunately not for all. In particular, I want to [pass a custom dispatcher to undici fetch](https://github.com/nodejs/undici/tree/main?tab=readme-ov-file#undicifetchinput-init-promise) so I can set the body timeout to infinite. This is necessary for listening to a `StreamingHTTPChannel2023` stream.

Please advise, thank you!
